Xamarin.Essentials/Types/Location.shared.cs

```diff
@@ -104,9 +112,11 @@ public Location(Location point)
 
         public override string ToString() =>
             $"{nameof(Latitude)}: {Latitude}, {nameof(Longitude)}: {Longitude}, " +
-            $"{nameof(Altitude)}: {Altitude ?? 0}, {nameof(Accuracy)}: {Accuracy ?? 0}, " +
-            $"{nameof(VerticalAccuracy)}: {VerticalAccuracy ?? 0}, " +
-            $"{nameof(Speed)}: {Speed ?? 0}, {nameof(Course)}: {Course ?? 0}, " +
+            $"{nameof(Altitude)}: " + Altitude != null ? $"{Altitude}" : string.Empty + ", " +
```

Comment:
I think that you can can simplify here and other places that use this null check.
```suggestion
            $"{nameof(Altitude)}: " + Altitude  ?? string.Empty + ", " +
```

Reply:
Thanks for the suggestion, @pictos, but this does not seem to work. For your code above, I get the error:

```
Error CS0019: Operator '??' cannot be applied to operands of type 'string' and 'bool' (CS0019) (Xamarin.Essentials)
```

This is a bit strange, and probably due to a missing parenthesis. If I add parentheses like this:

```
$"{nameof(Altitude)}: " + (Altitude ?? string.Empty) + ", " +
```

then I get:

```
Error CS0019: Operator '??' cannot be applied to operands of type 'double?' and 'string' (CS0019) (Xamarin.Essentials)
```

This is the same error I already saw for James' suggestion above (which was similar to yours).

Reply:
Yeah, this is odd, I'll take a look into the code to see if we are missing something. Before I suggest it, I [tried here](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4AEBMAjALABQBADGQQMwAEA3mXS3Qfo8697oRXQAc6AXjoAiKAHswcOgDcIAG3RwxAbi7cWABQQBLTDG0AKAJQbSW1kNGZ0ixRas79hk+c1bPPACx1dBkZmIgB83s68AJzGQgD8seJ6UHR2DmIelqwAvmRZQA=)

Reply:
@janusw I see the problem, it's because the `Altitude` is `double` and `string.Empty` is a `string`... To make it works you can use this approach if it's ok to use `null` as a value. You can play with this [here](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4AEBMAjALABQBADGQQMwAEA3mXS3Qfo869wG4QJ0IdALx1McAO50AMgHsAxhBgBLWZgAUFAHQUkhHQEou3Fk1ImLdAIIAbFTHQATOCLHobN4yYC+Abi/cAay4hACc6hBaACqyAMowCMqYAObqBgb+5qzeZDmkZF649HCY6GB0ACLKsBCY8nAAqpjKMFBeZpZ0ANLKNrJgcPAIUEhBLACyvXBtWSx5hcWl5bb2TnAASnAAZogl9SHtY3RNUAAOcPLKW8pwjq66R1GICBBJroSjsyYAoh7Kp1BZMo7qJ8J9OgBxOBAkF0GjgyyxdAILYQequAAsXnmXyKbA4ckUKjUhy+3DxhKUqg0RjJrA6lhxnQpCipanUjlk6GANhcNipDmcSDonO5vLofRSLTWtM6DM6LGkArWrn5q2cmQVrDkUsFLlEkuS0o1R24UWUAxqYFOrgqSjg5oGAHktlsoIMtA0YPIAHKyCSaxkFOksFlE6kcrk8vnKoUiqPiw3GuDCu3wR1wF1uwZ0FSWmAQa2yyzyhVK9X6iWxuCBrU6o161VqBtrWsKjNWm2iPPTAvWtvcJmWMNsjSi6NVivC8eJ5vJ6cJlwQOzJ4sWUudcvJ1XVgeWevbg1zvV7iwrQ+CFcn02sDt9ruVe0ZrPumCe71+gNHIcWEfEjSUv+dCnECmAwGuJgbpYVx0OoIFJDAIiiKUHgQVqLAwAAFgg/piJI1gIMkZQlDAPruDY3wAB71Kc/7qJghbQlscGgeBGTBuhMjVq48FgVoW7XiGJgHo2oi8W+ImtjeLB3oWD5pg6FpwO+vr+qeJjnqJwGsVomlSUJ3BWPI8jImiACePE6UZJkvPIZnqdwABqiAqIoNjWaZdmWQhWjOQgrnLh5tn2dJdCxOctzeXx4VwLcDmsAAwlywyVuJWhJci7rxSwACSUAAGI4WA4wKAA1gACjhPDAogUVvnlhX9CV8gVVVNUIOpP4mHiCnPq6r50LJ1qMHQySDL4dCvhNXXkvQM4uAJKoMKN42TatM3BHNi4yMeS0rTAE1TXQG2hltYpwAA/NYV57WNB1rfdJ1sGd0ZXUF5kjXdh3rRxw4vbyV1+QF7nGZ5FnLV9D3Tb9v7/ZdYURXcEOrUdT14vNV0ZSln0oz9+RCXiwCyLINh0A1RXNa1sjVc4AjI/dqMw919B6c4mw7AgexwCE10Vuzux1NzhA4wzeNHHiIQAGzxuddAJcuJlqnAVQ1ILkay2qybxPwMALhru3ONr/nCoB1ISqy/7fJgjiptUBaC00LRQHQWBOwYIgAHyhfLNiK/aKv2/U6ia3qRu6+buprGHwp9OGahW44/HVjHFvUgn/EGymLvNK07H48y9BSzLE4+378AB7UQem2o5tx5g0fF4m1YJ3rE5JnqLeVHbleNDnzuu7nnvSaX7j+93aux6OYdJ1OtdTwW/kZ5HcYh2snft2v1vCgPUB5+LheENL81ywro/l+PVepzXk//g31eYHPltb13qv1I7rTZ27Q8GYlp9KxXE8r71wXhJZOj9qTT0kivIB6dFrQLrrAzO28+57wJgfI+20R7/wvnAdQoVj6r0NiAhEWoCGZ2jvg7ahC4At0ofrZeNCt6hQAW/Pun9c5HCghYKAEgWjyEwrBHeaESyhW4Iod0L9A69ydloHofQBhDCgCAURw4ADsdAKDZRMOIlwLDpGtC0JMXkSiVG/nUZo0xIptgQHcKgSx3AsI4SkOIKQVhCLETAk6dAMAXTrFqGNKiNE6IMQGLIZiQi86cTyJ0NG9BqbPBqmwQgFBBpxASEkVI7thBex/iwAAJAAIgYCEpi6g4FwAMN4EAjByneGFMUxiYSymZ0qdUhgUC4B1LoAUugABqUKhSGmhOYqzCpVTGCjIutENJiQUhpDoBdK6IRtDfGtDAMy3gem9O6cKApbZvBAA)

`$"{nameof(Altitude)}: {Altitude ?? null}" + ", " +`

or 

` $"{nameof(Altitude)}: {Altitude.ToString() ?? string.Empty}" + ", "`
or 
` $"{nameof(Altitude)}: {Altitude?.ToString() ?? string.Empty}" + ", "`

Reply:
Ok, your last variant works well. But it turns out one can also just use ...

```
$"{nameof(Altitude)}: {Altitude}, "
```

... which has the desired effect of printing an empty string if Altitude is null. :)

Reply:
Now the ToString override looks amazing.